### PR TITLE
Upgrade to flutter_lints 2.0

### PR DESCRIPTION
**Pull request by goderbauer**
Requires https://github.com/flutter/packages/pull/1165 to be published. (Tests will fail until that has happened.)

### CODE REVIEW

**Comment by christopherfujino** [comment]
No code changes?

**Comment by goderbauer** [comment]
> No code changes?

I was hoping that the repro was already clean from my clean-up a couple of months ago, but looks like the checks are proving me wrong. Will look at those failures.

**Comment by goderbauer** [comment]
skp_generator is getting fixed in https://github.com/flutter/tests/pull/127.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: dev/bots/analyze_sample_code.dart [hunk 1]
```diff
@@ -746,9 +746,10 @@ dependencies:
     sdk: flutter
   flutter_test:
     sdk: flutter
+  vector_math:
```
Comment (christopherfujino) [inline dev/bots/analyze_sample_code.dart:749]: Can you make this explicit?
```suggestion
  vector_math: any
```

Comment (goderbauer) [reply dev/bots/analyze_sample_code.dart:749]: Will do!

**Comment by goderbauer** [comment]
This is currently blocked on https://github.com/dart-lang/linter/issues/3308.

**Comment by goderbauer** [comment]
The remaining failures should all get resolved as soon as the fix for https://github.com/dart-lang/linter/issues/3308 rolls into flutter.

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 7.
View them at https://flutter-gold.skia.org/cl/github/99881

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 7.
View them at https://flutter-gold.skia.org/cl/github/99881

**Comment by goderbauer** [comment]
The Dart roll went in, tests are passing. This is ready now.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**